src/main/scala/com/mozilla/telemetry/streaming/CrashesToInflux.scala

```diff
@@ -0,0 +1,192 @@
+      required = true)
+
+    conflicts(kafkaBroker, List(from, to, fileLimit, maxParallelRequests))
+
+    verify()
+  }
+
+  def parsePing(message: Message, channels: List[String], appNames: List[String],
+                measurementName: String): Array[String] = {
+    val fields = message.fieldsAsMap
+
+    if (!fields.get("docType").contains("crash")) {
+      Array[String]()
+    } else {
+      val ping = CrashPing(message)
+      val metadata = ping.meta
+
+      if (!channels.contains(metadata.normalizedChannel) || !appNames.contains(metadata.appName)) {
+        Array[String]()
+      } else {
+        val timestamp = metadata.Timestamp
+
+        val influxFields = ListMap(
+          "buildId" -> ping.getNormalizedBuildId.getOrElse(metadata.appBuildId)
+        )
+
+        val influxTags = ListMap(
+          "submissionDate" -> metadata.submissionDate,
+          "appVersion" -> metadata.appVersion,
+          "appName" -> metadata.appName,
+          "displayVersion" -> ping.getDisplayVersion.getOrElse(""),
+          "channel" -> metadata.normalizedChannel,
+          "country" -> metadata.geoCountry,
+          "osName" -> metadata.os.getOrElse(""),
+          "osVersion" -> ping.getOsVersion.getOrElse(""),
+          "architecture" -> ping.getArchitecture.getOrElse("")
+        )
+
+        val outputString = measurementName +
+          influxTags.map { case (k, v) => s"$k=$v" }.mkString(",", ",", "") +
+          influxFields.map { case (k, v) => s"$k=$v" }.mkString(" ", ",", "") +
+          " " +
+          timestamp
+
+        Array[String](outputString)
+      }
+    }
+  }
+
+  def getParsedPings(pings: DataFrame, raiseOnError: Boolean, measurementName: String,
+                     channels: List[String] = defaultChannels,
+                     appNames: List[String] = defaultAppNames): Dataset[String] = {
+    import pings.sparkSession.implicits._
+
+    pings.flatMap( v => {
+      try {
+        parsePing(Message.parseFrom(v.get(0).asInstanceOf[Array[Byte]]), channels, appNames, measurementName)
+      } catch {
+        case _: Throwable if !raiseOnError => Array[String]()
+      }
+    }).as[String]
+  }
+
+  def sendStreamCrashes(spark: SparkSession, opts: Opts): Unit = {
+    val httpSink = new RawHttpSink(opts.url(), Map())
+
+    val pings = spark
+      .readStream
+      .format("kafka")
+      .option("kafka.bootstrap.servers", opts.kafkaBroker())
+      .option("failOnDataLoss", opts.failOnDataLoss())
+      .option("kafka.max.partition.fetch.bytes", 8 * 1024 * 1024) // 8MB
+      .option("spark.streaming.kafka.consumer.cache.maxCapacity", kafkaCacheMaxCapacity)
+      .option("subscribe", TelemetryKafkaTopic)
+      .option("startingOffsets", opts.startingOffsets())
+      .load()
+
+    getParsedPings(pings.select("value"), opts.raiseOnError(),
+      opts.measurementName(), opts.acceptedChannels(), opts.acceptedAppNames())
+      .writeStream
+      .queryName(QueryName)
+      .foreach(httpSink)
```

Comment:
I still haven't figured out how to properly batch the calls together in streaming.  Any help would be appreciated

Reply:
I think we can handle that in ForeachWriter (copy current implementation of HttpSink: https://github.com/mozilla/telemetry-streaming/blob/master/src/main/scala/com/mozilla/telemetry/sinks/HTTPSink.scala#L30-L44) by:
    1) allocating empty array in `open`
    2) appending events there in `process`
    3) sending collected events in `close`

Another option is implementing `org.apache.spark.sql.execution.streaming.Sink`, which might offer more robust deduplication, but it's much more complicated and I think we don't need to overoptimize on this at this stage.